spec/system/awesome_map_spec.rb

```diff
@@ -13,6 +13,7 @@
   let!(:accepted_proposal) { create(:proposal, :accepted, component: proposal_component, latitude: 40, longitude: -50) }
   let!(:evaluating_proposal) { create(:proposal, :evaluating, component: proposal_component, latitude: 30, longitude: 45) }
   let!(:not_answered_proposal) { create(:proposal, :not_answered, component: proposal_component, latitude: 70, longitude: 6) }
+  let!(:null_state_proposal) { create(:proposal, nil, component: proposal_component, latitude: 50, longitude: 10) }
```

Comment:
```suggestion
  let!(:null_state_proposal) { create(:proposal, nil, component: proposal_component, latitude: 50, longitude: 10) }
```
This "nil" here won't work, symbols in rspec factories are "traits" that need to be registered. Passing "nil" is like doing nothing, it will create the default proposal.

This should work:
```suggestion
  let!(:null_state_proposal) { create(:proposal, state: nil, component: proposal_component, latitude: 50, longitude: 10) }
```

Reply:
Then, later on this file, you will want to make sure that the `null_state_propsal` is really displayed as a mark, this is done in the lines 111-114:
```ruby
  context "when step settings are all true" do
    it "shows all proposals markers" do
      sleep(1)
      expect(page.body).to have_selector("div[title='#{accepted_proposal.title["en"]}']")
      expect(page.body).to have_selector("div[title='#{evaluating_proposal.title["en"]}']")
      expect(page.body).to have_selector("div[title='#{rejected_proposal.title["en"]}']")
      expect(page.body).to have_selector("div[title='#{withdrawn_proposal.title["en"]}']")
      # check null are there:
      expect(page.body).to have_selector("div[title='#{null_state_propsal.title["en"]}']")
    end
  end
```